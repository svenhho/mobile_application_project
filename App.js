import React from 'react';
import AppNavigator from './components/AppNavigator';
import { NavigationContainer } from '@react-navigation/native';
import { LogBox } from 'react-native';
LogBox.ignoreLogs(['Warning: ...']); // Ignore log notification by message
LogBox.ignoreAllLogs();//Ignore all log notifications

export default function App() {

  console.disableYellowBox = true;

  return (
    <NavigationContainer>
      <AppNavigator />
    </NavigationContainer>
  );
}
