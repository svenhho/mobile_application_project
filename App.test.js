import React from 'react';
import { render } from '@testing-library/react-native';
import App from './App';
import renderer from 'react-test-renderer';

// Use this instead with React Native >= 0.64
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

describe('<App />', () => {
  test('simple check to see if it renders', () => {
    const tree = (
        <App />
    );
    
    render(tree);

    expect(tree).toBeTruthy();
  });


  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
