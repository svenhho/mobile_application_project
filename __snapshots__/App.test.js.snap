// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Login"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            behaviour="padding"
            onLayout={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#191414",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "700",
                }
              }
            >
              Login
            </Text>
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "700",
                }
              }
            >
              Please sign in to continue.
            </Text>
            <View
              style={
                Object {
                  "width": "80%",
                }
              }
            >
              <TextInput
                onChangeText={[Function]}
                placeholder="Email"
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderRadius": 10,
                    "marginTop": 5,
                    "paddingHorizontal": 15,
                    "paddingVertical": 10,
                  }
                }
                value=""
              />
              <TextInput
                onChangeText={[Function]}
                placeholder="Password"
                secureTextEntry={true}
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderRadius": 10,
                    "marginTop": 5,
                    "paddingHorizontal": 15,
                    "paddingVertical": 10,
                  }
                }
                value=""
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginTop": 40,
                  "width": "60%",
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#1db954",
                    "borderRadius": 10,
                    "opacity": 1,
                    "padding": 15,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 16,
                      "fontWeight": "700",
                    }
                  }
                >
                  Log in
                </Text>
              </View>
            </View>
            <View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "blue",
                      "fontSize": 16,
                      "textDecorationLine": "underline",
                    }
                  }
                >
                  Sign up
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
